ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryBuilder.java

```diff
@@ -446,33 +447,41 @@ PersistentQueryMetadata buildPersistentQueryInSharedRuntime(
         serviceContext
     );
 
-    return new BinPackedPersistentQueryMetadataImpl(
-        persistentQueryType,
-        statementText,
-        querySchema,
-        sources.stream().map(DataSource::getName).collect(Collectors.toSet()),
-        planSummary,
-        applicationId,
-        topology,
-        sharedKafkaStreamsRuntime,
-        runtimeBuildContext.getSchemas(),
-        config.getOverrides(),
-        queryId,
-        materializationProviderBuilder,
-        physicalPlan,
-        getUncaughtExceptionProcessingLogger(queryId),
-        sinkDataSource,
-        listener,
-        queryOverrides,
-        scalablePushRegistry,
-        (streamsRuntime) -> getNamedTopology(
-            streamsRuntime,
-            queryId,
+    final BinPackedPersistentQueryMetadataImpl binPackedPersistentQueryMetadata
+        = new BinPackedPersistentQueryMetadataImpl(
+            persistentQueryType,
+            statementText,
+            querySchema,
+            sources.stream().map(DataSource::getName).collect(Collectors.toSet()),
+            planSummary,
             applicationId,
+            topology,
+            sharedKafkaStreamsRuntime,
+            runtimeBuildContext.getSchemas(),
+            config.getOverrides(),
+            queryId,
+            materializationProviderBuilder,
+            physicalPlan,
+            getUncaughtExceptionProcessingLogger(queryId),
+            sinkDataSource,
+            listener,
             queryOverrides,
-            physicalPlan
-        )
+            scalablePushRegistry,
+            (streamsRuntime) -> getNamedTopology(
+                streamsRuntime,
+                queryId,
+                applicationId,
+                queryOverrides,
+                physicalPlan
+            )
     );
+    if (real) {
+      return binPackedPersistentQueryMetadata;
+    } else {
+      return SandboxedBinPackedPersistentQueryMetadataImpl.of(
```

Comment:
the new validation query was a normal query in  a sandbox env which was weird. This should prevent that